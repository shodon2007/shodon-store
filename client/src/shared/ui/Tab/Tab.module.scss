.tab {
	width: 100%;
	display: grid;
	grid-template-columns: 1fr 1fr;
	font-size: var(--font-size-m);
}

.btn.selected {
	color: var(--bg-color);
}

.btn:not(.selected) {
	background-color: transparent;
	color: var(--primary-color);
}
