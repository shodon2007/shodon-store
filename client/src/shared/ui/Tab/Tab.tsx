import { FC } from 'react';

import classNames from 'src/shared/lib/classNames/classNames';
import Button from 'src/shared/ui/Button/Button';

import cls from './Tab.module.scss';

interface TabProps {
	tabs: string[];
	tab: string;
	changeTab: React.Dispatch<React.SetStateAction<string>>;
}

const Tab: FC<TabProps> = props => {
	const { tabs, tab, changeTab } = props;

	function setTab(newValue: string) {
		changeTab(newValue);
	}

	return (
		<div className={cls.tab}>
			{tabs.map(elem => {
				return (
					<Button
						className={classNames(cls.btn, {
							[cls.selected]: tab === elem,
						})}
						key={elem}
						onClick={() => setTab(elem)}
					>
						{elem}
					</Button>
				);
			})}
		</div>
	);
};

export default Tab;
