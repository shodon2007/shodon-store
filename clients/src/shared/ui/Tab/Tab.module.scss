.tab {
	width: 100%;
	display: grid;
	grid-template-columns: 1fr 1fr;
	font-size: 16px;
}

.btn {
	background-color: transparent;
	color: white;
}

.btn.selected {
	color: #1a2733;
	background-color: white;
}
