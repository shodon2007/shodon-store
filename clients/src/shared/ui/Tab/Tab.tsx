import { FC } from "react"

import classNames from "/shared/lib/classNames";

import cls from './Tab.module.scss';

interface TabProps {
    tabs: string[];
    tab: string;
    changeTab: React.Dispatch<React.SetStateAction<string>>;
}

const Tab: FC<TabProps> = (props) => {
    const {
        tabs,
        tab,
        changeTab
    } = props;

    function setTab(newValue: string) {
        changeTab(newValue);
    }

    return (
        <div className={cls.tab}>
            {props.tabs.map(elem => {
                return <button className={classNames(cls.btn, { [cls.selected]: tab === elem })} key={elem} onClick={() => setTab(elem)}>{elem}</button>
            })}
        </div>
    )
}

export default Tab;