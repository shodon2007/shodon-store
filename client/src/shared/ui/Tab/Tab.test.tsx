import { render, screen } from '@testing-library/react';
import Tab from './Tab';

describe('testing Tab.tsx', () => {
	test('testing Tabs length', () => {
		render(
			<Tab
				changeTab={() => false}
				activeTab={'login'}
				tabs={{
					login: {
						name: 'войти',
						component: <div></div>,
					},
					registration: {
						name: 'зарегестрироваться',
						component: <div></div>,
					},
				}}
			/>,
		);

		const tab = screen.getByTestId('tab');
		const buttons = screen.getAllByTestId('button');

		expect(buttons.length).toBe(2);
		expect(tab).toMatchSnapshot();
	});
});
