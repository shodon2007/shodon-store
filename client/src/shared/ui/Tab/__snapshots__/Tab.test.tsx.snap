// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Tab.tsx testing Tabs length 1`] = `
<div
  class="tab"
  data-testid="tab"
>
  <button
    class="Button btn selected"
    data-testid="button"
  >
    войти
  </button>
  <button
    class="Button btn"
    data-testid="button"
  >
    зарегестрироваться
  </button>
</div>
`;
